CMakeLists.txt

```diff
@@ -27,18 +27,34 @@ project(Mir)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
-set(MIR_VERSION_MAJOR 0)
-set(MIR_VERSION_MINOR 29)
-set(MIR_VERSION_PATCH 0)
+if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/mir-version)
+  include(mir-version)
+  # When we create a tarball we need to generate contents like:
+  #  set(MIR_VERSION_MAJOR 0)
+  #  set(MIR_VERSION_MINOR 29)
+  #  set(MIR_VERSION_PATCH 0)
+  #  set(MIR_VERSION ${MIR_VERSION_MAJOR}.${MIR_VERSION_MINOR}.${MIR_VERSION_PATCH})
+else()
+  execute_process(
+    COMMAND git describe --tags --abbrev=0
```

Comment:
Oh yeah, and you don't want `--tags` here, we only want to consider annotated/signed tags.

Reply:
This is going to fail horribly when people try to build from snapshot URLs from GitHub.

Reply:
Well, it doesn't work today, either, due to submodule(s)...